crates/subspace-service/src/lib.rs

```diff
@@ -546,7 +546,9 @@ where
                             .expect("Must not be poisoned here")
                             .take()
                         {
-                            node_address_sender.send(address.clone()).unwrap();
+                            node_address_sender
+                                .send(address.clone())
+                                .expect("Channel must allow sending at this point.");
```

Comment:
Not really. Outer scope is an async function and can be dropped in the meantime, in which case receiver will be dropped, sending will error and this will cause panic propagating though networking stack and likely crashing the node as the result.

Reply:
Added `debug` logging.